.github/workflows/bench.yml

```diff
@@ -0,0 +1,49 @@
+name: Benchstat
+
+on:
+  pull_request:
+    branches: [ master ]
+
+permissions:  # added using https://github.com/step-security/secure-workflows
+  contents: read
+
+jobs:
+
+  bench:
+    name: Performance regression check
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout (previous)
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+          path: previous
+
+      - name: Checkout (new)
```

Comment:
Are you cloning twice?

You could clone once and switch branches.

Reply:
It only fetches a single commit by default. If I wanted to clone once and switch, I'd have to fetch all history for all branches and tags. This seems simpler/cleaner.